app/controllers/products_controller.rb

```diff
@@ -58,6 +59,10 @@ def edit
   private
 
   def product_params
-    params.require(:product).permit(:name, :description, :price, images: [])
+    params.require(:product).permit(:name, :description, :category_id, :price, images: [])
+  end
+
+  def set_categories
+    @categories = Category.all.all.order(:name)
```

Comment:
We dont need to order our categories by name it's useless, we can implement ordering fearure separately if you wish

Reply:
sorting categories by name is the best solution in this case, since sorting by something else (for example, the number of products in a category) is simply meaningless, since the user in 99% of cases simply does not need this information